### PR TITLE
Remove cairo-based backends from backend fallback.

### PR DESCRIPTION
**Pull request by anntzer**
They would never be selected by the fallback machinery anyways.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
Do we know why qt tests are failing on Travis?

**Comment by jklymak** [comment]
I guess I don't understand why they would never be selected?

File: matplotlibrc.template [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 ## The default backend. If you omit this parameter, the first
 ## working backend from the following list is used:
-## MacOSX Qt5Agg Qt4Agg Gtk3Agg GTK3Cairo TkAgg WxAgg Agg Cairo
+## MacOSX Qt5Agg Qt4Agg Gtk3Agg TkAgg WxAgg Agg
 ##
 ## Other choices include: WX PS PDF SVG Template.
```
Comment (jklymak) [inline matplotlibrc.template]: "other choices should now include the Cairos?

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -190,8 +190,8 @@ def switch_backend(newbackend):
     close("all")
 
     if newbackend is rcsetup._auto_backend_sentinel:
-        for candidate in ["macosx", "qt5agg", "qt4agg", "gtk3agg", "gtk3cairo",
-                          "tkagg", "wxagg", "agg", "cairo"]:
+        for candidate in ["macosx", "qt5agg", "qt4agg", "gtk3agg", "tkagg",
+                          "wxagg", "agg"]:
```
Comment (jklymak) [inline lib/matplotlib/pyplot.py]: maybe a comment in here why we won't fallback to the cairos?

**Comment by anntzer** [comment]
done x2

File: matplotlibrc.template [hunk 3]
```diff
@@ -31,9 +31,10 @@
 
 ## The default backend. If you omit this parameter, the first
 ## working backend from the following list is used:
-## MacOSX Qt5Agg Qt4Agg Gtk3Agg GTK3Cairo TkAgg WxAgg Agg Cairo
+## MacOSX Qt5Agg Qt4Agg Gtk3Agg TkAgg WxAgg Agg
 ##
-## Other choices include: WX PS PDF SVG Template.
+## Other choices include:
+## Qt5Cairo Qt5Cairo GTK3Cairo TkCairo WxCairo Wx PS PDF SVG Template.
```
Comment (timhoffm) [inline matplotlibrc.template]: Cairo?

Comment (anntzer) [reply matplotlibrc.template]: done

File: matplotlibrc.template [hunk 4]
```diff
@@ -31,9 +31,10 @@
 
 ## The default backend. If you omit this parameter, the first
 ## working backend from the following list is used:
-## MacOSX Qt5Agg Qt4Agg Gtk3Agg GTK3Cairo TkAgg WxAgg Agg Cairo
+## MacOSX Qt5Agg Qt4Agg Gtk3Agg TkAgg WxAgg Agg
 ##
-## Other choices include: WX PS PDF SVG Template.
+## Other choices include:
+## Qt5Cairo Qt5Cairo GTK3Cairo TkCairo WxCairo Cairo Wx PS PDF SVG Template.
```
Comment (QuLogic) [inline matplotlibrc.template]: Qt5Cairo appears twice; not sure if something else should go there.

Comment (anntzer) [reply matplotlibrc.template]: Sorry, am being sloppy; fixed.

**Pull request merged**